contracts/vaults/VoltzVault.sol

```diff
@@ -18,146 +22,38 @@ contract VoltzVault is IVoltzVault, IntegrationVault {
     using SafeCastUni for int128;
     using SafeCastUni for uint256;
     using SafeCastUni for int256;
+    using PRBMathSD59x18 for int256;
+    using PRBMathUD60x18 for uint256;
 
     IMarginEngine public _marginEngine;
     int24 _tickSpacing;
+    uint256 _leverage;
+    uint256 _k;
 
-    OpenedPositions _openedPositions;
     TickRange _currentPosition;
+    uint256 _currentPositionLiquidity;
+    TickRange[] trackedPositions;
```

Comment:
what happens when you rebalance to some old position? E.g. rebalance [1%,3%] -> then [3%, 5%] and the again to [1%, 3%]. 

What happens is that you try to sell all three positions. However, you try to settle one multiple times and the call fails -> withdrawal never works in that case and all funds are going to be blocked. handling previous positions needs more thinking...

What I'd also do is to put some flag in options for _pull operations (something like, onlyCurrent) that you can set to true in emergency cases and skips settling the previous positions.

Moreover, rebalancing might lead to long list of previous positions. What if settling all at once exceeds the gas limit? -> you're gonna end up in the case of never being able to call withdrawals...    One solution would be to add options of settling previous positions in batches.